internals/overlord/checkstate/manager.go

```diff
@@ -633,3 +644,50 @@ func (m *CheckManager) Replan() {
 		m.updateCheckData(check, changeID, 0)
 	}
 }
+
+// RefreshCheck runs a check immediately.
+func (m *CheckManager) RefreshCheck(ctx context.Context, check *plan.Check) (*CheckInfo, error) {
+	m.checksLock.Lock()
+	checkData := m.ensureCheck(check.Name)
+	refresh := checkData.refresh
+	result := checkData.result
+	m.checksLock.Unlock()
+
+	getCheckInfo := func() *CheckInfo {
+		m.checksLock.Lock()
+		checkData := m.ensureCheck(check.Name)
+		m.checksLock.Unlock()
+		info := CheckInfo{
+			Name:      checkData.name,
+			Level:     checkData.level,
+			Startup:   checkData.startup,
+			Status:    checkData.status,
+			Failures:  checkData.failures,
+			Threshold: checkData.threshold,
+			ChangeID:  checkData.changeID,
+		}
+		return &info
+	}
+
+	// If the check is stopped, run the check directly without using changes and tasks.
+	if checkData.changeID == "" {
```

Comment:
I think we need to fetch the `changeID` here with the lock held, just like we're fetching the `refresh` channel.